EIPS/eip-4469.md

```diff
@@ -0,0 +1,55 @@
+---
+eip: 4469
+title: Wrapped metaverse token
+description: Defines a standard interface for viewing the addresses and balances of wrapped metaverse tokens.
+author: Hyungsuk Kang (@hskang9)
+discussions-to: https://ethereum-magicians.org/t/eip-4469-wrapping-erc1155-into-erc20s/7516
+status: Draft
+type: Standards Track
+category: ERC
+created: 2021-11-21
+---
+
+## Abstract
+
+Metaverses(e.g. Decentraland, SandBox, etc) are now emerging in Ethereum ecosystem, and the standard of the tokens in each metaverse are all aggregated into ERC-1155 for efficiency.
+Invented by Witek Radomski from Enjin, this has made many metaverse/game project to accomodate their assets to provide ownerships. 
+However, current ERC-1155 has been sacrificing its interoperability to other universes and current defi ecosystem due to different data structure and mechanics. A certain contract to bridge them into prepared ERC-20 could be possible, but this just can create duplicate datas in the global state if source is not uniform. Due to this, I propose the standard wrapped metaverse token contract to wrap metaverse assets into the most widely used ERC-20 tokens. 
+
+This EIP introduces a simple interface for exposing the addresses of all wrapped ERC-1155 tokens in metaverses, as well as querying wrapped assets' balances.
+
+## Motivation
+Many metaverse(or game) projects are choosing ERC-1155 for efficiently generating their assets. Choosing ERC-1155 for managing non-fungible tokens or other tokens in the game has brought memory efficiencies and less contract address for systems to operate. However, this has brought incompatibility between other existing defi infrastructure(e.g. Dexes, Lending, etc). When a user wants to liquidate their game assets or metaverse assets, he or she needs to change it to intermediary assets to acquire it then change it to ETH then buy other assets in different metaverse. This is very inefficient for interoperability between connecting ecosystems between metaverses as well due to gas fees required to convert one asset with another. Just as Wrapped ether has been invented, I propose to have wrapped ERC-1155 asset that can be both backward compatible to existing decentralized finance infrastructure and free trade between metaverses.
+
+### Example
+
+For example, if a user wants to sell their assets from existing metaverse, he or she wraps through the wrapper 
+
+## Specification
+Using `create2`, Wrapped metaverse tokens must be generated from creating wrapped ERC-20 from the owner of ERC-1155.
```

Comment:
Why is `create2` necessary here?

Reply:
Is there a way to store all erc1155 data in one ERC20 contract?

Reply:
without `create2`, how does one create each corresponding wrapped erc20 contract?

Reply:
I might not understand what you mean by "corresponding", but why wouldn't a `create` ERC20 contract also work?

Reply:
That would also work, and yeah I should look into it.

Reply:
I would not choose `create` because it does not give predictable address that the wrapper can navigate wrapped ERC1155->ERC20 token. Choosing `create` just because it creates a contract is a dangerous idea.

Reply:
This will require the wrapper contract to keep the list of addresses then gas price will also increase.